app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/FahrplanFragment.java

```diff
@@ -493,7 +493,11 @@ private void setBell(Session session) {
     private void scrollTo(@NonNull Session session) {
         final ScrollView parent = requireViewByIdCompat(getView(), R.id.scrollView1);
         int height = getNormalizedBoxHeight(getResources(), scale, LOG_TAG);
-        final int pos = (session.relStartTime - conference.getFirstSessionStartsAt()) / 5 * height;
+        int startsAtMinuteUtc = session.relStartTime - conference.getFirstSessionStartsAt();
+        int systemOffsetMinutes = Moment.getSystemOffsetMinutes();
+        // Translate start time minutes from UTC to system time zone rendered to the user.
+        int startsAtMinuteSystem = startsAtMinuteUtc - systemOffsetMinutes;
+        final int pos = startsAtMinuteSystem / 5 * height;
```

Comment:
Not sure how the UI and positioning works. More consistent and easy would be to have everything internally as UTC (or datetime with offset, doesnt matter actually) and then convert to "system local" as needed.

So instead of calculating startsAdMinuteUtc I'd rather have the firstSessionStart as absolute UTC datetime and the sessionStart also as datetime. Then you'd get something like a TimeSpan object when subtracting those and take the total minutes from that.

However, the approach in this commit will also work as intended - except for the days that have the actual DST change (this already happened on easterhegg14 or sometime around). But I guess that there will be additional problems with rendering, especially when clock is set back and one hour "appears twice".

Reply:
I agree that it would be easier to have all calculation wrapped in the `Moment` class. I already added a `TODO` [here](https://github.com/EventFahrplan/EventFahrplan/pull/344/commits/845b25a6d18c53c86993b761dbf9fb6eb6f9ee30) but did not take time to work on it.

Reply:
I prepared a beginning to encapsulate the calculation in the `Moment` class:

``` java
Moment sessionStartsAtMoment = Moment.ofEpochMilli(session.dateUTC);
Moment firstSessionStartsAtMoment = Moment.ofEpochMilli(conference.getFirstSessionStartsAt());
Moment startsAtMoment = sessionStartsAtMoment.minus(firstSessionStartsAtMoment);
```

I am not sure how to to implement the `minus` function best - here is a draft:

``` kotlin
operator fun minus(moment: Moment): Moment {
    return ofEpochMilli(time.minus(moment.time.toEpochMilli(), ChronoUnit.MILLIS).toEpochMilli())
}
```

When it comes to apply the system time offset I haven't found the Java 8 resp. ThreeTenBp API yet. Any hints are appreciated.

Reply:
I think it is sufficient to calculate in granularity of minutes, like take the minutes and multiply them with 60.000. 

Regarding the minus, one of the three objects involved probably needs to have another data type. You cannot subtract two datetimes and get a datetime. Either you subtract two datetimes and get a time span, or you subtract a timespan from a datetime and get another datetime.
Maybe you're trapped into having the "startsAtMoment" variable as a moment. Actually what you want there is a plain number of time units that tells you the exact physical duration between the first session and the current session. this can by nature not be a moment, it's either a number of seconds,minutes,whatever or a Timespan (Instant?). So your minus function must not return a moment but just the milliseconds - or even minutes so you dont need to change anything somewhere else. The important thing is only that you have both times "proper" before subtracting them with a proper library - then you get the real number of minutes between them and scrolling (or even rendering a timetable) will be correct.